src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.TextTemplating/TextTemplatingStep.cs

```diff
@@ -1,28 +1,25 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
-using Microsoft.DotNet.Scaffolding.Helpers.Services;
-using Microsoft.DotNet.Scaffolding.TextTemplating;
-using Microsoft.Extensions.Logging;
+using Microsoft.DotNet.Scaffolding.Core.Scaffolders;
+using Microsoft.DotNet.Scaffolding.Core.Steps;
 
-namespace Microsoft.DotNet.Scaffolding.Helpers.Steps;
+namespace Microsoft.DotNet.Scaffolding.TextTemplating;
 /// <summary>
 /// Provided all the required properties, this ScaffoldStep can template T4 onto a file on disk.
 /// </summary>
-internal class AddTextTemplatingStep : ScaffoldStep
+public class TextTemplatingStep : ScaffoldStep
 {
     //Path to .tt (T4) template on disk (likely to be packed in a dotnet tool)
-    public required string TemplatePath { get; init; }
+    public required string TemplatePath { get; set; }
     //the System.Type auto-generated object of the template (using TextTemplatingFilePreprocessor)
-    public required Type TemplateType { get; init; }
-    //output file path where the templated content will be written
-    public required string OutputPath { get; init; }
+    public required Type TemplateType { get; set; }
+    //output file path where the templated content will be written (should include the extension if one is wanted)
+    public required string OutputPath { get; set; }
     //the 'name' property of <#@ parameter #> in provided .tt template
-    public required string TemplateModelName { get; init; }
+    public required string TemplateModelName { get; set; }
     //the 'type' property of <#@ parameter #> in provided .tt template
-    public required object TemplateModel { get; init; }
-    public required IFileSystem FileSystem { get; init; }
```

Comment:
These should be able to be imported still since the step comes from DI. Just add a constructor to TextTemplatingStep that takes them as parameters. Then you can undo some of the other changes in this file I think.

Reply:
but IFileSystem comes from the `Helpers` library which is not referenced in this project? I was avoiding adding that reference until our dependency stuff is sorted (next PR)

Reply:
Ah, yeah, you're right about IFileSystem. ILogger should be fine though. Maybe leave a TODO comment about the FileSystem so we don't forget and put ILogger back in?

Reply:
(note the import in the constructor here would be `ILogger<TextTemplatingStep>`, not just `ILogger`, though you can store it as an ILogger if you want, it doesn't really matter other than for import.

Reply:
yep I already have those changes stashed actually, was avoiding adding it in this PR and will do it together for all logging.